tests/test_preprocess.py

```diff
@@ -27,6 +28,8 @@ def setup_method(self):
         self.df_rab_polys = gpd.GeoDataFrame(
             geometry=[g for g in plgns], crs=self.df_streets_rabs.crs
         )
+        test_file_path3 = mm.datasets.get_path("nyc_graph", extension="graphml")
+        self.graph = ox.get_undirected(ox.load_graphml(test_file_path3))
```

Comment:
Not sure exactly what is going on, but it appears to be an issue with `ox.load_graphml()`. Can you confirm that the file path in `test_file_path3` is correct?

Reply:
The file path is corrected according to the rewriting of the get_path function to deal with other extensions. Is it possible that osmnx is not included in the minimal test environment?

Reply:
hmmm. `osmnx` is in the [minimal](https://github.com/pysal/momepy/blob/afcb85b89195f3ba2c5c869fb2e5f5add357c927/ci/envs/38-minimal.yaml#L14) testing environment.

Reply:
But it does seems likely that some dependency is missing in that environment and is causing the error. Or a dependency that is needed is not supported by Python 3.8. If we look [here](https://github.com/gboeing/osmnx/blob/main/osmnx/io.py#L252) within `osmnx`, the `.read_graphml()` is being called from `networkx` with keyword `force_multigraph=True`, which we can see in the CI error log [here](https://github.com/pysal/momepy/runs/8294778205?check_suite_focus=true#step:6:1289).

Reply:
So after some more digging we can see that the version of `networkx` being pulled in for the Python 3.8 test is [v2.3](https://github.com/pysal/momepy/runs/8294778205?check_suite_focus=true#step:5:157), which was [released in 2019](https://github.com/networkx/networkx/releases/tag/networkx-2.3). And that version of `networkx` did not support the `force_multigraph` keyword as in does now (see [v.2.3 `GraphMLReader`](https://github.com/networkx/networkx/blob/807d9fde187e18a58d542f87ab0d0cb98ef46459/networkx/readwrite/graphml.py#L237) vs. [current `main`'s `GraphMLReader`](https://github.com/networkx/networkx/blob/034274e08b3b78022e47d03cc5e69e28be23f713/networkx/readwrite/graphml.py#L293)).

So the trick now is to find the earliest version of `networkx` that supports `force_multigraph` in the `GraphMLReader` signature and [update the pin in the CI environment](https://github.com/pysal/momepy/blob/afcb85b89195f3ba2c5c869fb2e5f5add357c927/ci/envs/38-minimal.yaml#L8).

Reply:
And it looks like that will be [`networkx=2.5`](https://github.com/networkx/networkx/blob/12f852f6a686ed32f718125c5965586dd6bb82a9/networkx/readwrite/graphml.py#L199).

Reply:
Thanks a lot, James! I wouldn't have been able to figure this out. Updated it and let's see if it goes through now